src/constructs/cloudwatch/ec2-alarms.ts

```diff
@@ -0,0 +1,50 @@
+import { ComparisonOperator, MathExpression, TreatMissingData } from "@aws-cdk/aws-cloudwatch";
+import { HttpCodeElb, HttpCodeTarget } from "@aws-cdk/aws-elasticloadbalancingv2";
+import type { GuStack } from "../core";
+import type { AppIdentity } from "../core/identity";
+import type { GuApplicationLoadBalancer } from "../loadbalancing";
+import type { GuAlarmProps } from "./alarm";
+import { GuAlarm } from "./alarm";
+
+/**
+ * Creates an alarm which is triggered whenever the percentage of requests with a 5xx response code exceeds
+ * the specified threshold.
+ */
+export interface Gu5xxPercentageMonitoringProps
```

Comment:
This is the input we need from the user

Reply:
Not sure if you've seen how AWS CDK implement this? They have a [`private` directory](https://github.com/aws/aws-cdk/tree/master/packages/cdk-assets/lib/private) for private interfaces/types. That is, if something is defined outside there, it's a public interface. Not sure if we can cleanly copy that pattern (or even if we should?).

Reply:
Interesting, I hadn't looked into that before. 

> Not sure if we can cleanly copy that pattern (or even if we should?).

I'm not sure about this part either; will have a think about it!